js/ui/windowManager.js

```diff
@@ -2069,19 +2076,43 @@ const CodingManager = new Lang.Class({
         this._firstFrameConnections.push(firstFrameConnection);
     },
 
-    _windowAppPositionChanged: function(window, event, session) {
+    _windowAppPositionChanged: function(window, session) {
         let rect = session.windowApp.get_meta_window().get_frame_rect();
         session.buttonApp.set_position(rect.x + rect.width - BUTTON_OFFSET_X, rect.y + rect.height - BUTTON_OFFSET_Y);
         if (!session.windowBuilder)
-                return;
+            return;
+        session.windowBuilder.get_meta_window().move_resize_frame(false,
+                                                                  rect.x,
+                                                                  rect.y,
+                                                                  rect.width,
+                                                                  rect.height);
+    },
+
+    _windowAppSizeChanged: function(window, session) {
+        let rect = session.windowApp.get_meta_window().get_frame_rect();
+        session.buttonApp.set_position(rect.x + rect.width - BUTTON_OFFSET_X, rect.y + rect.height - BUTTON_OFFSET_Y);
+        if (!session.windowBuilder)
+            return;
         session.windowBuilder.get_meta_window().move_resize_frame(false,
                                                                   rect.x,
                                                                   rect.y,
                                                                   rect.width,
                                                                   rect.height);
     },
```

Comment:
I don't know if it makes sense to duplicate this function. Why not just used the same function for both?